portfolio/src/test/java/com/google/sps/ReviewDaoTest.java

```diff
@@ -0,0 +1,163 @@
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableSet;
+import com.google.sps.data.Book;
+import com.google.sps.data.Review;
+import com.google.sps.data.ReviewDao;
+import com.google.sps.data.ReviewDaoDatastore;
+import java.lang.IllegalAccessException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+   This is a test class for BookDao.java
+    Tests the implementations ability to create,
+    parse, put, retrieve, update, and delete
+    entities in Datastore
+*/
+
+@RunWith(JUnit4.class)
+public final class ReviewDaoTest {
+  private ReviewDao reviewDao = new ReviewDaoDatastore();
+  private static final String DEFAULT_ISBN_1 = "1111111111111";
+  private static final String DEFAULT_ISBN_2 = "1111111111112";
+  private static final String DEFAULT_EMAIL_1 = "unknown@email.com0";
+  private static final String DEFAULT_EMAIL_2 = "unknown@email.com1";
+  private static final String DEFAULT_EMAIL_3 = "unknown@email.com2";
+  private static final String USER_EMAIL_1 = "user1@br.com";
+  private static final String USER_EMAIL_2 = "user2@br.com";
+
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(
+      new LocalDatastoreServiceTestConfig().setApplyAllHighRepJobPolicy());
+
+  @Test
+  public void uploadAllTest() {
+    Book bookWithReviews = createBook();
+    reviewDao.uploadAll(bookWithReviews);
+    Set<Review> actual = reviewDao.getAllByISBN(DEFAULT_ISBN_1);
+    Set<Review> expected =
+        new HashSet<>(Arrays.asList(Review.create("Review #1", DEFAULT_ISBN_1, DEFAULT_EMAIL_1),
+            Review.create("Review #2", DEFAULT_ISBN_1, DEFAULT_EMAIL_2),
+            Review.create("Review #3", DEFAULT_ISBN_1, DEFAULT_EMAIL_3)));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void uploadNewNormalTest() throws Exception {
+    Review review1 = Review.create("New user review #1", DEFAULT_ISBN_1, USER_EMAIL_1);
+    Review review2 = Review.create("New user review #2", DEFAULT_ISBN_2, USER_EMAIL_1);
+    reviewDao.uploadNew(review1);
+    reviewDao.uploadNew(review2);
+    Set<Review> actual = reviewDao.getAllByEmail(USER_EMAIL_1);
+    Set<Review> expected = new HashSet<>(Arrays.asList(review1, review2));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test(expected = Exception.class)
+  public void uploadNewEdgeTest() throws Exception {
```

Comment:
nit: instead of saying `uploadNewEdgeTest`, you could name the method more descriptive, something like `uploadNew_reviewAlreadyExists` .